AGENTS.md

```diff
@@ -0,0 +1,375 @@
+## Who This Is For
+
+- **AI agents**: Automate repository tasks with minimal context
+- **Contributors**: Humans using AI assistants or working directly
+- **Maintainers**: Ensure assistants follow project conventions and CI rules
+
+## What This Document Provides
+
+- Environment setup and canonical commands for format, lint, and tests
+- Repository map and conventions to keep changes consistent
+- Guardrails for PRs, CI, and releases
+- Quick references for common tasks and troubleshooting
+
+## Project Overview
+
+**Purpose**: Kubeflow SDK provides a unified Python SDK for AI practitioners to interact with multiple Kubeflow projects via consistent APIs, focusing on user workflows over infrastructure details.
+
+**Problem It Solves**: Reduces Kubernetes and multi-project complexity, offering simple, local-first Python interfaces for training, tuning, pipelines (planned), and model lifecycle management.
+
+**Key Benefits**:
+- Unified experience across Kubeflow projects
+- Simplified AI workflows with minimal infrastructure knowledge
+- Local development support (install via `pip`) with optional cluster backends
+
+**Today's Scope**:
+- **Available**: Kubeflow Trainer (train/fine-tune with different backends)
+- **Planned**: Katib (HPO), Pipelines (workflows), Model Registry
+- See README "Supported Kubeflow Projects" for current status
```

Comment:
Will this help AI Agents to write/debug code or we can remove this to reduce context ?